gcp_variant_transforms/beam_io/vcf_header_io.py

```diff
@@ -28,10 +29,63 @@
 
 from gcp_variant_transforms.beam_io import vcfio
 
-__all__ = ['VcfHeader', 'VcfHeaderSource', 'ReadAllVcfHeaders',
-           'ReadVcfHeaders', 'WriteVcfHeaders']
+__all__ = ['HeaderConflictResolver', 'VcfHeader', 'VcfHeaderSource',
+           'ReadAllVcfHeaders', 'ReadVcfHeaders', 'WriteVcfHeaders']
 
 
+class HeaderConflictResolver(object):
+  """ A class for resolving mistmatch between header fields
+      (e.g. type, num).
+  """
+
+  # List of all resolvable conflicts with their resolution.
+  # For example, entry [a, b, c] means mistmatch between a and b is resolved to
+  # c (where c is usually either a or b).
+  _RESOLVABLE_CONFLICTS = [
+      # Conflict between 'Integer' and 'Float' is resolved to 'Float'.
+      ['Integer', 'Float', 'Float'],
```

Comment:
I love the extensibility of this class, but if we don't add any more resolvable conflicts to this list it might be more complex than is necessary.

Reply:
Asha mentioned another type of conflicts we want to resolve, e.g., 'num' fields with any value greater than 1 are compatible. I plan to add regex to this list for this case (in a separate cl). Apart from these two cases, I am not sure if there is more.